### PR TITLE
CBL-4223: Analysis Warnings (3/13) LiteCore/Query

### PR DESCRIPTION
**Pull request by callumbirks**
No notable changes. Mostly just optimizing imports, adhering to language standards, marking unused parameters, etc.

### CODE REVIEW

File: LiteCore/Query/SQLiteFleeceEach.cc [hunk 1]
```diff
@@ -176,7 +177,7 @@ namespace litecore {
 
 #pragma mark - INSTANCE METHODS:
 
-        FleeceCursor(FleeceVTab* vtab) : _vtab(vtab) {}
+        explicit FleeceCursor(FleeceVTab* vtab) : sqlite3_vtab_cursor{vtab}, _vtab(vtab) {}
```
Comment (borrrden) [inline LiteCore/Query/SQLiteFleeceEach.cc:180]: What is the righthand side of this change doing?

File: LiteCore/Query/SQLiteFleeceUtil.cc [hunk 2]
```diff
@@ -150,7 +156,9 @@ namespace litecore {
                 case kNumber:
                     if ( val->isInteger() )
                         if ( val->isUnsigned() ) {
-                            sqlite3_result_int64(ctx, val->asUnsigned());
+                            // val->asUnsigned() does cause some overflow error with narrow_cast, so we can't use it
```
Comment (borrrden) [inline LiteCore/Query/SQLiteFleeceUtil.cc:159]: Actually in release builds it does not, but this is fine as well

**Review by borrrden — APPROVED** [review]

**Review by jianminzhao — APPROVED** [review]

File: LiteCore/Query/SQLiteFleeceEach.cc (hunk 1 above)
Comment (callumbirks) [reply LiteCore/Query/SQLiteFleeceEach.cc:180]: Initializes the base class `sqlite3_vtab_cursor`. 
There is another constructor for the object:
`explicit FleeceCursor(FleeceVTab* vtab) : sqlite3_vtab_cursor{vtab}, _vtab(vtab) {}`
Which initializes both the base class, and `_vtab`.
Interestingly, `_vtab` has the same data type as the base class, so it's possible we only need one of them. Going to investigate that now.

Comment (callumbirks) [reply LiteCore/Query/SQLiteFleeceEach.cc:180]: Hmm yeah, removing the inheritance from `sqlite3_vtab_cursor`, although it causes no compile errors, causes fatal errors during QueryTest. So I guess some sqlite3 functions that takes pointers etc, rely on the passed object inheriting from sqlite3_vtab_cursor.

File: LiteCore/Query/SQLiteFleeceUtil.cc (hunk 2 above)
Comment (callumbirks) [reply LiteCore/Query/SQLiteFleeceUtil.cc:159]: 😮

**Pull request merged**